Framework/Core/src/AnalysisDataModel.cxx

```diff
@@ -0,0 +1,105 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "Framework/AnalysisDataModel.h"
+#include "Framework/StringHelpers.h"
+#include "Framework/Logger.h"
+#include <boost/algorithm/string.hpp>
+
+namespace o2::aod::datamodel
+{
+std::string getTreeName(header::DataHeader dh)
+{
+  std::string description = std::string(dh.dataDescription.str);
+  std::string origin = std::string(dh.dataOrigin.str);
+
+  // lower case of first part of description
+  auto found = description.find_first_of(":");
+  std::string treeName = boost::algorithm::to_lower_copy(description).substr(0, found);
+
+  // add prefix according to origin
+  if (origin == "AOD") {
+    treeName = "O2" + treeName;
+  } else if (origin == "RN2") {
+    treeName = "Run2" + treeName;
+  }
+
+  // exceptions from this
+  if (origin == "AOD" && description == "MCCOLLISLABEL") {
+    treeName = "O2mccollisionlabel";
+  }
+
+  return treeName;
+}
+
+template <typename... C>
+static constexpr auto columnNamesTrait(framework::pack<C...>)
+{
+  return std::vector<std::string>{C::columnLabel()...};
+}
+
+std::vector<std::string> getColumnNames(header::DataHeader dh)
+{
+  auto description = std::string(dh.dataDescription.str);
+  auto origin = std::string(dh.dataOrigin.str);
+
+  // get column names
+  // AOD / RN2
+  if (origin == "AOD") {
+    if (description == "TRACK:PAR") {
+      return columnNamesTrait(typename StoredTracksMetadata::table_t::persistent_columns_t{});
+    } else if (description == "TRACK:PARCOV") {
+      return columnNamesTrait(typename StoredTracksCovMetadata::table_t::persistent_columns_t{});
+    } else if (description == "TRACK:EXTRA") {
+      return columnNamesTrait(typename TracksExtraMetadata::table_t::persistent_columns_t{});
```

Comment:
After a discussion with Paul, my understanding is that these lines are only needed for tables where we read the table partially (at the moment the Track table). The other lines are not needed, but don't harm either. I am worried that they may give the impression that new tables will not be supported.
@ktf @aalkin What is your opinion?

Reply:
This is an unfortunate consequence of matching runtime information to static types, one way or another you need to hardcode this correspondence between the two. Additionally, this may allow us to have versioning for trees/tables that we are reading, which requires to have a list of branches/columns known, similar to partially read trees.

Reply:
My understanding is that we only need to hardcode it for Tracks.
My question is if we shouldn't drop it for all other tables, where we do *not* need to hardcode it. At least the code also works for user-provided tables without changing the AOD reader.

Reply:
Indeed. For whatever does not have `:` in the name, we can simply assume that the columns are the ones which are found in the tree, no?

Reply:
I was also thinking that in the long term we can remove it also for the partial tables (e.g. TRACKs), by having a static linked list (a C one, not an std::list) were for each element we have the pointer to the next element, the label for the table (e.g. TRACKS:PAR) and a pointer to a function which returns the list of columns for that table. This should allow us to avoid any hardcoding.

Reply:
@ktf Shall we then keep only the Track lines, the rest should be handled by the default. And if we find a `:` in the name but it is not handled explicitly, this could throw a fatal error

Reply:
@pbuehler Could you do this modification, please? Many thanks! Then we merge it.